core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -129,14 +133,24 @@ public LogicalPlan analyze(UnresolvedPlan unresolved, AnalysisContext context) {
   @Override
   public LogicalPlan visitRelation(Relation node, AnalysisContext context) {
     QualifiedName qualifiedName = node.getTableQualifiedName();
+    Set<String> allowedCatalogNames = catalogService.getCatalogs()
+        .stream()
+        .map(Catalog::getName)
+        .collect(Collectors.toSet());
     CatalogSchemaIdentifierName catalogSchemaIdentifierName
-        = new CatalogSchemaIdentifierName(qualifiedName.getParts(), catalogService.getCatalogs());
+        = new CatalogSchemaIdentifierName(qualifiedName.getParts(), allowedCatalogNames);
```

Comment:
what is the CatalogSchemaIdentifierName of `.CATALOGS`? 
catalog = .opensearch
schema = default
table = .catalogs
if yes, `show catalogs` equal to `select * from .opensearch.default.catalogs`.  should it be handled by `.opensearch` storage engine?
or we consider `select * from .catalogmeta.info_schema.catalogs`.

to be clarify, this is not a blocker of this PR.

Reply:
With current implementation, it is `select * from .opensearch.default.catalogs`. But it doesn't make sense now.


But when we start storing catalogs in index in future.
It will be meta info under Opensearch storageEngine.  Then `.opensearch.information_schema.catalogs` makes sense.